pulp_container/app/models.py

```diff
@@ -200,6 +200,48 @@ class Meta:
         unique_together = (("name", "tagged_manifest"),)
 
 
+class ImageManifestSignature(Content):
+    """
+    A signature for an image manifest.
+
+    Fields:
+        manifest_digest (models.CharField): A manifest digest as specified in signature metadata.
+        identity_reference (models.CharField): An identity reference as specified in signature
+                                               metadata. Usually a full tag, but can be a digest.
+                                               Short versions of tags are not allowed.
+        signature_type (models.CharField): A signature type as specified in signature metadata.
+                                           Currently it's only "atomic container signature".
+        signing_key_id (models.CharField): A key id identified by gpg (last 8 bytes of the
+                                           fingerprint).
+        signature_timestamp (models.PositiveIntegerField): A signature timestamp identified by gpg.
+
+
+    Relations:
+        signed_manifest (models.ForeignKey): An image manifest this signature is relevant to.
+
+    """
+    PROTECTED_FROM_RECLAIM = False
+
+    TYPE = "signature"
+
+    manifest_digest = models.CharField(max_length=255, db_index=True)
+    identity_reference = models.CharField(max_length=255)
+    signature_type = models.CharField(max_length=255)  # maybe make it a choice field with one
+                                                       # choice so far "atomic"
+    signing_key_id = models.CharField(max_length=255, db_index=True)
+    signature_timestamp = models.PositiveIntegerField()
+
+    signed_manifest = models.ForeignKey(
```

Comment:
can we also add `creator`?

Reply:
I learned more about this field and we propbably will not gain much by exposing it on the model - it identifies the name and version of the software which has created the signature, not the top-level caller so in our case it will aclways be 'atomic' since we use skopeo https://github.com/containers/skopeo/blob/99621f416892746bd8d223fa5c4773267124e9cd/vendor/github.com/containers/image/v5/signature/signature.go#L70

Reply:
maybe it's the case with skopeo signing but other signatures may contain something more useful. E.g. from rh registry I get `{"creator": "Red Hat RCM Pub 0.67.0-final"}`

Reply:
I'm fine with leaving it out as well.